iis/iis_test.go

```diff
@@ -278,50 +278,68 @@ func TestWebsite(t *testing.T) {
 		t.Fatal("Error purging: ", err)
 	}
 
-	// Basic config to be used for creating a website
-	config := &TaskConfig{
+	websiteConfig := &WebsiteConfig{
+		Name: guid,
 		Path: "C:\\inetpub\\wwwroot",
+		Bindings: []iisBinding{
+			{Type: "http", Port: 8080},
+			{Type: "https", Port: 8081, CertHash: hash},
+		},
+		Env: map[string]string{
+			"EXAMPLE_ENV_VAR":     "test123",
+			"EXAMPLE_ENV_VAR_ALT": "test123",
+		},
 		AppPoolIdentity: iisAppPoolIdentity{
 			Identity: "SpecificUser",
 			Username: "vagrant",
 			Password: "vagrant",
 		},
-		Bindings: []iisBinding{
-			{Type: "http", Port: 8080},
-			{Type: "https", Port: 8081, CertHash: hash},
-		},
 		AppPoolConfigPath: "C:\\vagrant\\vagrant\\testapppool.xml",
 		SiteConfigPath:    "C:\\vagrant\\vagrant\\testsite.xml",
 	}
 
 	// Create a website with the config and website name
-	if err := createWebsite(guid, config); err != nil {
+	if err := createWebsite(websiteConfig); err != nil {
 		t.Fatal(err)
 	}
```

Comment:
I know we discussed this in the past (and understand that this was not easy to accomplish with weird windows CI setup), but these tests should be incorporated as part of CI. With tests running as part of CI, we can be confident that the PR is working and has no regressions. This is especially important now since we are getting community PRs.

Some details here https://app.gitkraken.com/glo/card/9a9e4b88684c4cbeaa64e1d01070b7a2

Reply:
I agree, but won't be a part of the PR.